frontend/src/components/TimerPanel/TimerPanel.tsx

```diff
@@ -0,0 +1,215 @@
+import { Controller, type SubmitHandler, useForm } from "react-hook-form"
+import FormRow from "../Common/FormRow.tsx"
+
+import {
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+  Checkbox,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+} from "@chakra-ui/react"
+import { useMutation, useQueryClient } from "@tanstack/react-query"
+import dayjs from "dayjs"
+import type { ApiError, TDataCreateSchedule } from "../../client"
+import { createSchedule } from "../../client/services/scheduleService.ts"
+import useCustomToast from "../../hooks/useCustomToast.ts"
+import { handleError } from "../../utils.ts"
+
+interface TimerFormData {
+  startDate: string
+  endDate?: string
+  daysBetween: number
+  skipWeekends: boolean
+  skipHolidays: boolean
+  timeOfDay: string
+}
+const TimerPanel = () => {
+  // eslint-disable-next-line
+  const today: string = dayjs().format("YYYY-MM-DD")
+  // eslint-disable-next-line
+  const tomorrow: string = dayjs().add(1, "day").format("YYYY-MM-DD")
+  // eslint-disable-next-line
+  const aMonthFromToday: string = dayjs().add(1, "month").format("YYYY-MM-DD")
+
+  // eslint-disable-next-line
+  const { handleSubmit, register, control, watch, setValue } = useForm({
+    defaultValues: {
+      startDate: tomorrow,
+      endDate: aMonthFromToday,
+      daysBetween: 1,
+      timeOfDay: "08:00",
+      skipHolidays: false,
+      skipWeekends: false,
+    },
+  })
+
+  // eslint-disable-next-line
+  const endDate = watch("endDate")
+  // eslint-disable-next-line
+  const startDate = watch("startDate")
+
+  const showToast = useCustomToast()
+  const queryClient = useQueryClient()
+  const mutation = useMutation({
```

Comment:
see this is an example of business logic inside a view. But not sure, perhaps this is how it's supposed to be done in react. @Sergei-S-V ?

Reply:
(granted, there's no actual "logic" per se -- it's more around the operation / what to do on success/failure.

Reply:
sure. it's a good recommendation. I will pull this out into a hook.